File: apps/ios/Barns/BarnsApp/Domain/UseCases/MyItems/GetProductItemDetailUseCase.swift
/// Returns a single registered product item by id, if present.
struct GetProductItemDetailUseCase {
    private let repository: ProductItemRepository

    init(repository: ProductItemRepository) {
        self.repository = repository
    }

    func execute(id: ProductItem.ID) async throws -> ProductItem? {
        try await repository.productItem(id: id)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/myitems/GetProductItemDetailUseCase.kt
package com.barns.app.domain.usecase.myitems

import com.barns.app.domain.model.ProductItem
import com.barns.app.domain.repository.ProductItemRepository

/**
 * Returns a single registered product item by id, if present.
 */
class GetProductItemDetailUseCase(
    private val repository: ProductItemRepository,
) {
    suspend fun execute(id: String): ProductItem? = repository.productItem(id)
}
